config/templating/templating_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package templating
+
+import (
+	"testing"
+)
+
+func TestHasTemplating(t *testing.T) {
+	tests := []struct {
+		vars          []string
+		hasTemplating bool
+	}{
+		{
+			vars:          []string{"foo=bar"},
+			hasTemplating: false,
+		},
+		{
+			vars:          []string{"foo={public_IPv4}"},
+			hasTemplating: true,
+		},
+		{
+			vars:          []string{"foo=public_IPv4}"},
+			hasTemplating: true,
+		},
+		{
+			vars:          []string{"foo={public_IPv4"},
+			hasTemplating: true,
+		},
+		{
+			vars:          []string{"foo=}public_IPv4{"},
+			hasTemplating: true,
+		},
+	}
+	for i, test := range tests {
+		if test.hasTemplating != HasTemplating(test.vars) {
+			t.Errorf("#%d: hasTemplating didn't match", i)
+		}
+	}
+}
+
+func TestPerformTemplating(t *testing.T) {
+	tests := []struct {
+		platform     string
+		vars         []string
+		expectedVars []string
```

Comment:
Please use the testing structure from Ignition: https://github.com/coreos/ignition/blob/8e66c946c57cca732041b1489492a1295c1ed0b1/config/translate_test.go#L34